packages/store-sync/src/createStoreSync.ts

```diff
@@ -291,7 +292,7 @@ export async function createStoreSync<config extends StoreConfig = StoreConfig>(
             return { status, blockNumber, transactionHash };
           }
         } catch (error) {
-          if (error instanceof TransactionReceiptNotFoundError) {
+          if (isViemTransactionReceiptNotFoundError(error)) {
```

Comment:
can we use the strategy that viem recommends in https://viem.sh/docs/error-handling.html?